lib/entities/parse_address_from_domain.dart

```diff
@@ -17,57 +20,60 @@ const unstoppableDomains = [
   'blockchain'
 ];
 
-Future<ParsedAddress> parseAddressFromDomain(
-    String domain, String ticker) async {
-  try {
-    final formattedName = OpenaliasRecord.formatDomainName(domain);
-    final domainParts = formattedName.split('.');
-    final name = domainParts.last;
-
-    if (domainParts.length <= 1 || domainParts.first.isEmpty || name.isEmpty) {
-      try {
-        final addresses = await fetchYatAddress(domain, ticker);
-
+Future<ParsedAddress> parse(String text, String ticker) async {
+    try {
+        if (text.hasOnlyEmojis) {
+        final yatService = getIt.get<YatService>();
+        final addresses = await yatService.fetchYatAddress(text, ticker);
         if (addresses?.isEmpty ?? true) {
           return ParsedAddress(
-              addresses: [domain], parseFrom: ParseFrom.yatRecord);
+              addresses: [text], parseFrom: ParseFrom.yatRecord);
         }
-
         return ParsedAddress(
-            addresses: addresses, name: domain, parseFrom: ParseFrom.yatRecord);
-      } catch (e) {
-        return ParsedAddress(addresses: [domain]);
+            addresses: addresses.map((e) => e.address).toList(),
+            name: text,
+            parseFrom: ParseFrom.yatRecord,
+          );
       }
-    }
+      final formattedName = OpenaliasRecord.formatDomainName(text);
+      final domainParts = formattedName.split('.');
+      final name = domainParts.last;
 
-    if (unstoppableDomains.any((domain) => name.contains(domain))) {
-      final address = await fetchUnstoppableDomainAddress(domain, ticker);
+      if (domainParts.length <= 1 || domainParts.first.isEmpty || name.isEmpty) {
+        return ParsedAddress(addresses: [text]);
+      }
+
+      if (unstoppableDomains.any((domain) => name.contains(domain))) {
+        final address = await fetchUnstoppableDomainAddress(text, ticker);
+
+        if (address?.isEmpty ?? true) {
+          return ParsedAddress(addresses: [text]);
+        }
 
-      if (address?.isEmpty ?? true) {
-        return ParsedAddress(addresses: [domain]);
+        return ParsedAddress(
+            addresses: [address],
+            name: text,
+            parseFrom: ParseFrom.unstoppableDomains);
```

Comment:
Same here, please make another one method which will fetch address by domain and return parsed address

Reply:
done